web/src/message_list_view.ts

```diff
@@ -420,13 +434,14 @@ type SubscriptionMarkers = {
 function populate_group_from_message(
     message: Message,
     date_unchanged: boolean,
+    display_year: boolean,
```

Comment:
Maybe we should call this something like `year_changed`? I think that's better communicate what this information means.